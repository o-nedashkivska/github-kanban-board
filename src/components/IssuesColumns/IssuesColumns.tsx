// import useOctokit from "../../hooks/useOctokit";
import { Row } from "antd";
import IssueColumn from "../IssueColumn";

import styles from "./issues-columns.module.css";

const IssuesColumns: React.FC = () => {
  // const { data } = useOctokit("facebook", "react");
  // console.log(data, "data");

  return (
    <Row justify="space-evenly" className={styles.columns}>
      <IssueColumn title="ToDo" />
      <IssueColumn title="In Progress" />
      <IssueColumn title="Done" />
    </Row>
  );
};

export default IssuesColumns;
