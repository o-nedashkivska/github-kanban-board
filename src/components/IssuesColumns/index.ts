import IssuesColumns from "./IssuesColumns";

export default IssuesColumns;
