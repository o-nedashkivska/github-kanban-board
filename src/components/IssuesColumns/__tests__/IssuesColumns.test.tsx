import { screen } from "@testing-library/react";

import { renderWithProviders } from "../../../test-utils";

import IssuesColumns from "../IssuesColumns";

describe("IssuesColumns", () => {
  test("displays three columns with correct names", () => {
    renderWithProviders(<IssuesColumns />);

    expect(screen.getByText("ToDo")).toBeInTheDocument();
    expect(screen.getByText("In Progress")).toBeInTheDocument();
    expect(screen.getByText("Done")).toBeInTheDocument();

    expect(screen.getAllByText("No issues")).toHaveLength(3);
  });

  test("matches snapshot", () => {
    const { asFragment } = renderWithProviders(<IssuesColumns />);

    expect(asFragment()).toMatchSnapshot();
  });
});
