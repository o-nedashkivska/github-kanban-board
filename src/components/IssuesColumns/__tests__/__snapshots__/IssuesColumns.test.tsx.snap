// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IssuesColumns matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="ant-row ant-row-space-evenly columns css-dev-only-do-not-override-1kuana8"
  >
    <div
      class="ant-col ant-col-6 column css-dev-only-do-not-override-1kuana8"
    >
      <h4
        class="ant-typography column__title css-dev-only-do-not-override-1kuana8"
      >
        ToDo
      </h4>
      <div
        class="column__content ant-flex css-dev-only-do-not-override-1kuana8 ant-flex-align-center ant-flex-gap-middle ant-flex-vertical"
      >
        <h4
          class="ant-typography css-dev-only-do-not-override-1kuana8"
        >
          No issues
        </h4>
      </div>
    </div>
    <div
      class="ant-col ant-col-6 column css-dev-only-do-not-override-1kuana8"
    >
      <h4
        class="ant-typography column__title css-dev-only-do-not-override-1kuana8"
      >
        In Progress
      </h4>
      <div
        class="column__content ant-flex css-dev-only-do-not-override-1kuana8 ant-flex-align-center ant-flex-gap-middle ant-flex-vertical"
      >
        <h4
          class="ant-typography css-dev-only-do-not-override-1kuana8"
        >
          No issues
        </h4>
      </div>
    </div>
    <div
      class="ant-col ant-col-6 column css-dev-only-do-not-override-1kuana8"
    >
      <h4
        class="ant-typography column__title css-dev-only-do-not-override-1kuana8"
      >
        Done
      </h4>
      <div
        class="column__content ant-flex css-dev-only-do-not-override-1kuana8 ant-flex-align-center ant-flex-gap-middle ant-flex-vertical"
      >
        <h4
          class="ant-typography css-dev-only-do-not-override-1kuana8"
        >
          No issues
        </h4>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
